velox/connectors/hive/SplitReader.cpp

```diff
@@ -299,14 +304,25 @@ void SplitReader::setNullConstantValue(
 void SplitReader::setPartitionValue(
     common::ScanSpec* spec,
     const std::string& partitionKey,
-    const std::optional<std::string>& value) const {
+    const std::optional<std::string>& value,
+    bool isCastStringToDateIso8601) const {
   auto it = partitionKeys_.find(partitionKey);
   VELOX_CHECK(
       it != partitionKeys_.end(),
       "ColumnHandle is missing for partition key {}",
       partitionKey);
-  auto constValue = VELOX_DYNAMIC_SCALAR_TYPE_DISPATCH(
-      convertFromString, it->second->dataType()->kind(), value);
+  velox::variant constValue;
+  if (it->second->dataType()->isDate()) {
```

Comment:
Should this logic be part of convertFromString?

Reply:
Updated.